### DISCUSSION TITLE
No starship prompt under nix-shell?

### DISCUSSION THREAD

**Discussion by tmpm697** [starship/starship#5378] [Q&A]
i have this simple setup:
```
add_newline = false
format = "$time$nix_shell$git_status$directory$character"
scan_timeout = 10

[nix_shell]
disabled = false
```
but when i issue `nix-shell -p git` --> the shell prompt is just like this `[nix-shell:~/.config]$` --> this is not starship prompt 

why? and any working nix_shell portion?

**Comment by mickimnet** [comment]
IIRC, you need to set up your `shell.nix` so that it includes the _Starship_ prompt. Also, how did you install _Starship_ on _NixOS_? Because there is are _Starship_ options for _Nix_.

**Reply by tmpm697** [reply]
`programs.starship` under hm. thanks.
what `shell.nix` contain?

**Reply by mickimnet** [reply]
Sorry, my bad. I just check my setup again. I don't have it in `shell.nix` anymore.

What shell do you use normally? As `nix-shell` normally defaults to `bash`. I have `eval "$(starship init bash)"` in my `~/.bash_profile`.

**Reply by tmpm697** [reply]
ah, yes, it drop to `bash`, type `zsh` to switch shell does show `$nix_shell`, how to default it to zsh then?

**Reply by tmpm697** [reply]
I had `users.defaultUserShell = pkgs.zsh;`

**Reply by tmpm697** [reply]
an `alias` should work `nix-shell = "nix-shell --command zsh"`. Thanks.

**Reply by davidkna** [reply]
IIRC `nix shell` will keep you in the same shell, but in that case, you may need to enable `nix_shell.heuristic` to trigger the `nix_shell` module. To get starship in `bash`, you could also try enabling `programs.bash` in hm.

**Reply by tmpm697** [reply]
you mean smth will not work if I do `alias`?

**Reply by davidkna** [reply]
No, only if you use `nix shell` instead of `nix-shell`

**Comment by amok** [comment]
That's how nix works. You have to ether use it as `nix-shell --command zsh` or use something like https://github.com/haslersn/any-nix-shell or similar (if exists)
